### ISSUE TITLE
Enhancement: Support Json Maps

### ISSUE THREAD

**Issue by ecout** [godaddy/kubernetes-external-secrets#308]
Currently this project has a leg up above aws-secret-operator when it comes to AWS SM support when it comes to Plain Text mode.
The problem is that it is limited, the parsing of Map Objects is not complete.
I.e.
AWS SM - Plain Text:

```
{
  "item1": {
    "username": "that_user",
    "password": ""
  },
  "item2": {
    "domain": "etc.url.com",
    "region": "region2"
  },
  "item3": "ThisOneIsFlat",
  "item4": "IAmFlatToo"
}
```

If we use this secret definition:
```
apiVersion: kubernetes-client.io/v1
kind: ExternalSecret
metadata:
 name: my.individual.secrets
 namespace: "secrets"
spec:
 backendType: secretsManager
 data:
 - key: my/secret
 name: map
 property: item1
 - key: event/map
 name: flat
 property: item3
```

Or, if we use this one:
```
apiVersion: kubernetes-client.io/v1
kind: ExternalSecret
metadata:
 name: my.whole.secret
 namespace: "secrets"
spec:
 backendType: secretsManager
 dataFrom:
 - my/secret
 ```

We ALWAYS get: base64 encoded:
W29iamVjdCBPYmplY3Rd
For items with subitems like item1 and 2

Which is javascript generic Object type:
[object Object]

The following is a request for the engine to PARSE that object and subdivide into each component.
The KEY question being, DO KUBERNETES SECRETS support Maps?

Thanks!

**Comment by Flydiverny** [comment]
Believe this is solved here https://github.com/godaddy/kubernetes-external-secrets/pull/247

Sadly I can't merge it myself

**Comment by Flydiverny** [comment]
By solved I mean the object object part, not sure if this fulfills your request here.

**Comment by ecout** [comment]
Hello @silasbw, can you please REVIEW this Pull request so we can close this issue? Thanks!

https://github.com/godaddy/kubernetes-external-secrets/pull/247

**Comment by Flydiverny** [comment]
#247 is merged and should have been released in 3.2.0, please reopen if it didn't solve this! :)
